daemon/api_snaps.go

```diff
@@ -882,24 +960,51 @@ func meetSnapConstraintsForEnforce(ctx context.Context, inst *snapInstruction, s
 }
 
 func snapRemoveMany(_ context.Context, inst *snapInstruction, st *state.State) (*snapInstructionResult, error) {
-	flags := &snapstate.RemoveFlags{Purge: inst.Purge}
-	removed, tasksets, err := snapstateRemoveMany(st, inst.Snaps, flags)
-	if err != nil {
-		return nil, err
+	if len(inst.Snaps) == 0 && len(inst.CompsForSnaps) == 0 {
+		return nil, fmt.Errorf("cannot remove zero snaps")
 	}
 
-	var msg string
-	switch len(inst.Snaps) {
-	case 0:
-		return nil, fmt.Errorf("cannot remove zero snaps")
-	case 1:
-		msg = fmt.Sprintf(i18n.G("Remove snap %q"), inst.Snaps[0])
-	default:
-		quoted := strutil.Quoted(inst.Snaps)
-		// TRANSLATORS: the %s is a comma-separated list of quoted snap names
-		msg = fmt.Sprintf(i18n.G("Remove snaps %s"), quoted)
+	var compsTaskSets, snapsTaskSets []*state.TaskSet
+	var removed []string
+	var msg, snapsMsg, compsMsg string
+	var err error
+	if len(inst.CompsForSnaps) > 0 {
+		for snap := range inst.CompsForSnaps {
+			if strutil.ListContains(inst.Snaps, snap) {
+				return nil, fmt.Errorf(i18n.G("cannot mix components and snap when removing %q"), snap)
```

Comment:
from this message is not very clear that is means one can't remove the snap and components from at the same time